### PR TITLE
[Not Ready] Clothes Manufacturing

### PR DESCRIPTION
**Pull request by XElectricX**
Opening a PR to keep record of progress and take advantage of Travis for any errors.

### CODE REVIEW

**Comment by Zero-Bits** [comment]
**_# Textiles Intensify_**

**Comment by XElectricX** [comment]
I swear on my oath as a Singulo Knight, we WILL have sweatshops.

File: code/game/machinery/clothes crafting/clothes_crafter.dm [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+	for(var/T in materials)
+		. += list(list("mat" = capitalize(T), "amt" = materials[T]))
+
+/obj/machinery/clothes_crafter/proc/eject_materials(var/material, var/amount) // 0 amount = 0 means ejecting a full stack; -1 means eject everything
+	var/recursive = amount == -1 ? 1 : 0
+	material = lowertext(material)
+	var/mattype
+	switch(material)
+        if("cloth")
+			mattype = /obj/item/stack/material/cloth
```
Comment (HeyBanditoz) [inline code/game/machinery/clothes crafting/clothes_crafter.dm]: Fix the whitespace here. Or my phone formatted it wrong.

File: code/game/machinery/clothes crafting/clothes_crafter.dm [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+	if(!(material in materials))
+		to_chat(user, "<span class=warning>\The [src] does not accept [stack_plural]!</span>")
+		return
+
+	if(materials[material] + amnt <= res_max_amount)
+		if(stack && stack.amount >= 1)
+			var/count = 0
+			overlays += "fab-load-metal"
+			spawn(10)
```
Comment (HeyBanditoz) [inline code/game/machinery/clothes crafting/clothes_crafter.dm]: Use a timer if the codebase has it.

Comment (Zero-Bits) [reply code/game/machinery/clothes crafting/clothes_crafter.dm]: IIRC this was forked before timers were moved to bay

File: code/game/machinery/clothes crafting/clothes_crafter.dm (hunk 1 above)
Comment (Zero-Bits) [reply code/game/machinery/clothes crafting/clothes_crafter.dm]: Aye, there's an extra tab.

Comment (Zero-Bits) [reply code/game/machinery/clothes crafting/clothes_crafter.dm]: It's a broken whitespace.

**Pull request closed without merging**